src/Core/Aetherling/Passes/ThroughputModifications.hs

```diff
@@ -1,339 +1,508 @@
-module Aetherling.Passes.ThroughputModifications where
+{-|
+Module: Aetherling.Passes.ThroughputModifications
+Description: Passes that tradeoff throughput and area
+
+The Aetherling Operations. These are split into four groups:
+
+1. Leaf, non-modifiable rate - these are arithmetic, boolean logic,
+and bit operations that don't contain any other ops and don't have
+a parameter for making them run with a larger or smaller throughput.
+Since these don't have a modifiable rate, they are sped up and slowed
+down by wrapping them in a parent, modifiable rate op such as map and
+underutil.
+
+2. Leaf, modifiable rate - these are ops like linebuffers,
+and space-time type reshapers that have a parameter for changing their
+throughput and typically are not mapped over to change their throuhgput.
+These ops don't have child ops.
+Since these have a modifiable rate, they are sped up and slowed down by
+trying to adjust that rate. In some cases, it may not be possible to adjust
+the rate if the op with the new is invalid given the dimensions of the data
+being operated on. Speed up and slow down may fail in these cases.
+
+3. Parent, non-modifiable rate - these ops like composeSeq and composePar have
+child ops that can have their throughputs' modified, but the parent
+op doesn't have a parameter that affects throughput
+
+4. Parent, modifiable rate - map is the canonical example. It has child ops
+and can have its throughput modified by changing parallelism.
+
+The four groups are have their throughputs increased and decreased using
+different approaches:
+
+1. Leaf, non-modifiable rate - these ops are sped up and slowed down by wrapping
+them in a parent, modifiable rate op such as map and underutil. 
+
+2. Leaf, modifiable rate - these ops are sped up and slowed down by trying to
```

Comment:
"trying to adjust their rate"  --> trying to adjust their rate via changes to rate parameters.  Since not all ops in this category permit all possible rates, it may not be possible to speed up/slow down these ops in a cases.

<Meta: I'm thinking we could be a little more precise about use of the term "adjust their rate" in these comments, since we also talk about changing the rate of a non-modifiable rate op.  I get what we mean, but we might confuse someone if the context is less clear.>

Reply:
I agree that rate is not the best term. I've switched to the term "throughput parameter". I introduce the term https://github.com/David-Durst/aetherlingHaskellIR/pull/14/files#diff-a4e9be19161290150b0e024cdc796b07R7 and https://github.com/David-Durst/aetherlingHaskellIR/pull/14/files#diff-a4e9be19161290150b0e024cdc796b07R33. Thoughts? I believe this clears up the confusion.

I've also decreased the confusion by renaming the categories indirectly and directly scalable. See https://github.com/David-Durst/aetherlingHaskellIR/pull/14/files#diff-a4e9be19161290150b0e024cdc796b07R7 for their definitions.